xdp-flowradar_kern.c

```diff
@@ -93,31 +128,26 @@ struct five_tuple {
  */
 static __always_inline
 uint16_t hash(uint16_t host, uint8_t k, struct five_tuple *ft) {
+    unsigned long i;
 	uint32_t a = 63689;
 	uint32_t b = 378551;
 
-	uint16_t saddr1 = (uint16_t)(ft->saddr >> 16);
-	uint16_t saddr2 = (uint16_t)(ft->saddr & 0x0000ffff);
-	uint16_t daddr1 = (uint16_t)(ft->daddr >> 16);
-	uint16_t daddr2 = (uint16_t)(ft->daddr & 0x0000ffff);
+    char *ptr = (void *)ft;
 
 	uint32_t hash = k;
-	hash = hash * a + host;
-	a = a * b;
-	hash = hash * a + saddr1;
-	a = a * b;
-	hash = hash * a + saddr2;
-	a = a * b;
-	hash = hash * a + daddr1;
-	a = a * b;
-	hash = hash * a + daddr2;
-	a = a * b;
-	hash = hash * a + ft->sport;
-	a = a * b;
-	hash = hash * a + ft->dport;
-	a = a * b;
-	hash = hash * a + ft->proto;
+#pragma unroll
```

Comment:
Maybe this is a common C practice, but why are these loops specifically unrolled and not others?